src/main/java/org/complate/graal/GraalComplateRendererBuilder.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.complate.graal;
+
+import org.complate.core.ComplateSource;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Value;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.graalvm.polyglot.HostAccess.ALL;
+
+/**
+ * Builder with sane default that has to be used to create an instance of
+ * {@link GraalComplateRenderer}.
+ */
+public final class GraalComplateRendererBuilder {
+
+    private static final Supplier<Context.Builder> DEFAULT_CONTEXT_BUILDER_FACTORY =
+        () -> Context.newBuilder("js")
+            .allowHostAccess(ALL)
+            .allowHostClassLookup((s) -> true)
+            .allowExperimentalOptions(true)
+            .option("js.experimental-foreign-object-prototype", "true");
+    private static final Function<Context.Builder, Context.Builder> NO_OP_CUSTOMIZER =
+        Function.identity();
+
+    private Supplier<Context.Builder> contextBuilderFactory = DEFAULT_CONTEXT_BUILDER_FACTORY;
+    private Function<Context.Builder, Context.Builder> contextCustomizer = NO_OP_CUSTOMIZER;
+    private final Map<String, Object> bindings = new HashMap<>();
+    private final ComplateSource source;
+
+    GraalComplateRendererBuilder(ComplateSource source) {
+        this.source = source;
+    }
+
+    /**
+     * Adds the given bindings to the previous ones.
+     *
+     * @param bindings the bindings to add
+     * @return an {@link GraalComplateRendererBuilder} instance with all previous
+     *         state and this bindings on top
+     */
+    public GraalComplateRendererBuilder withBindings(Map<String, ?> bindings) {
+        this.bindings.putAll(bindings);
+        return this;
+    }
+
+    /**
+     * Adds the given binding to the previous ones.
+     *
+     * @param identifier the global identifier the value can be accessed within
+     *                   the views
+     * @param value      the value of this binding
+     * @return an {@link GraalComplateRendererBuilder} instance with all previous
+     *         state and this binding on top
+     */
+    public GraalComplateRendererBuilder withBinding(String identifier, Object value) {
+        this.bindings.put(identifier, value);
+        return this;
+    }
+
+    /**
+     * The given {@link Context.Builder} is used to create a {@link Context} instance
```

Comment:
You may want to say that the "sane defaults" promised in the class' javadocs no longer apply once I provide a `Context.Builder` of my own.